src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -967,56 +967,47 @@ internal static ErrorCode committed(IntPtr rk, IntPtr partitions, IntPtr timeout
 
-        private delegate ErrorCode Producev(IntPtr rk,
-                ProduceVarTag topicTag, string topic,
-                ProduceVarTag partitionTag, int partition,
-                ProduceVarTag vaTag, IntPtr val, UIntPtr len,
-                ProduceVarTag keyTag, IntPtr key, UIntPtr keylen,
-                ProduceVarTag msgflagsTag, IntPtr msgflags,
-                ProduceVarTag msg_opaqueTag, IntPtr msg_opaque,
-                ProduceVarTag timestampTag, long timestamp,
-                ProduceVarTag headersTag, IntPtr headers,
-                ProduceVarTag endTag);
-        private static Producev _producev;
         internal static ErrorCode producev(
-                IntPtr rk,
-                string topic,
-                int partition,
-                IntPtr msgflags,
-                IntPtr val, UIntPtr len,
-                IntPtr key, UIntPtr keylen,
-                long timestamp,
-                IntPtr headers,
-                IntPtr msg_opaque)
-            => _producev(rk,
-                    ProduceVarTag.Topic, topic,
-                    ProduceVarTag.Partition, partition,
-                    ProduceVarTag.Value, val, len,
-                    ProduceVarTag.Key, key, keylen,
-                    ProduceVarTag.Opaque, msg_opaque,
-                    ProduceVarTag.MsgFlags, msgflags,
-                    ProduceVarTag.Timestamp, timestamp,
-                    ProduceVarTag.Headers, headers,
-                    ProduceVarTag.End);
-
+            IntPtr rk,
+            string topic,
+            int partition,
+            IntPtr msgflags,
+            IntPtr val, UIntPtr len,
+            IntPtr key, UIntPtr keylen,
+            long timestamp,
+            IntPtr headers,
+            IntPtr msg_opaque)
+        {
+            IntPtr topicStrPtr = Marshal.StringToCoTaskMemAnsi(topic);
+            rd_kafka_vu[] vus =
+            {
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Topic,     data  = new rd_kafka_vu_data() {topic = topicStrPtr}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Partition, data  = new rd_kafka_vu_data() {partition = partition}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.MsgFlags,  data  = new rd_kafka_vu_data() {msgflags = msgflags}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Value,     data  = new rd_kafka_vu_data() {val = new rd_kafka_key_val() {data = val, size = len}}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Key,       data  = new rd_kafka_vu_data() {val = new rd_kafka_key_val() {data = key, size = keylen}}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Timestamp, data  = new rd_kafka_vu_data() {timestamp = timestamp}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Headers,   data  = new rd_kafka_vu_data() {headers = headers}},
+                new rd_kafka_vu() {vt = rd_kafka_vtype.Opaque,    data  = new rd_kafka_vu_data() {opaque = msg_opaque}},
+            };
+            try
```

Comment:
this can't throw unless i'm missing something.

Reply:
Not relevant any more